modules/openid_auth/app/controllers/openid_auth/v2/validation_controller.rb

```diff
@@ -81,6 +86,28 @@ def setup_structure
         end
         payload_object
       end
+
+      def charon_token_screen?(payload_object)
+        act_vista_id = payload_object.act[:vista_id]
+        sta3n = payload_object.launch['sta3n']
+        return false unless !act_vista_id.nil? && !sta3n.nil?
+
+        vista_ids = act_vista_id.scan(/\d{3}[A-Z]*\|\d+\^[A-Z]{2}\^\d{3}[A-Z]*\^[A-Z]{5}\|[A-Z]{1}/)
+        return false unless vista_ids
+
+        vista_ids.each do |vista_id|
+          parsed_sta3n = vista_id.match(/\d{3}|/)
+          no_match =  !sta3n.to_s.eql?(parsed_sta3n.to_s)
+          return true unless no_match
```

Comment:
Are these two lines (100, 101) driven by linting rules? The double inverted logic is quite hard to read. I had to break out `irb` to make sure I understood this.

`return true if sta3n.to_s.eql?(parsed_sta3n.to_s)` is a lot cleaner.